packages/ses/src/compartment-shim.js

```diff
@@ -83,13 +89,143 @@ const assertModuleHooks = compartment => {
   }
 };
 
-/**
- * Compartment()
- * The Compartment constructor is a global. A host that wants to execute
- * code in a context bound to a new global creates a new compartment.
+/** SharedCompartmentConstructor
+ *
+ * All compartments share a prototype, though there are many compartment
+ * constructors.
+ * The compartment shim introduces a fully powered global Compartment constructor.
+ * The lockdown shim replaces that Compartment constructor with one whose descendants
+ * all share the new tamed and hardened environment.
+ * Within each Compartment, there is a unique Compartment constructor that
+ * closes over the parent.
+ *
+ * Before lockdown, we expect every constructor to have a prototype and for the
+ * prototype to refer back to the constructor.
+ * After lockdown, we expect the constructors from all compartments to share a
+ * prototype, but the prototype can no longer carry a reference back to any
+ * particular constructor.
+ * For example, every compartment has a unique and powerful Function
+ * constructor, each of these has the same prototype.
+ * Because the prototype is shared among compartments, the shared constructor
+ * on the prototype must be powerless.
+ *
+ * Following this pattern, the shared compartment constructor is also powerless
+ * and simply throws an exception.
+ */
+export const SharedCompartmentConstructor = function Compartment() {
+  throw new TypeError(
+    `Cannot call or construct compartment with shared constructor`,
+  );
+};
+
+SharedCompartmentConstructor.toString = () => {
```

Comment:
If there's no hurry, I suggest this PR wait till after https://github.com/Agoric/SES-shim/pull/372 , or at least till after the relevant parts of #372 are somehow merged in. The relevant parts untangle some of the whitelist and intrinsic handling to that, for example, the powerful start `Date` and the powerless shared `Date.prototype.constructor` co-exist better. 

Btw, it also adds these non-std `toString` methods to the whitelist, without which they were being silently removed.